modules/pwa/pwa.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Plugin Name: PWA
+ * Description: Add Progressive Web App support to your WordPress site.
+ * Plugin URI: https://github.com/automattic/jetpack
+ * Author: Automattic
+ * Author URI: https://automattic.com
+ * Version: 0.4.2
+ * Text Domain: pwa
+ * Domain Path: /languages/
+ * License: GPLv2 or later
+ */
+
+ /**
+  * Include the following PWA capabilities:
+  * - cache the home page and posts/pages
+  * - cache all CSS and JS
+  * - show offline/online status using body class "jetpack__offline"
+  * TODO:
+  * - push updates
+  * - how to cache within wp-admin?
+  * - hook WP's native cache functions to expire and push updates to sites
+  */
+
+define( 'PWA_SW_QUERY_VAR', 'jetpack_service_worker' );
+define( 'PWA_MANIFEST_QUERY_VAR', 'jetpack_app_manifest' );
+
+// module activation
+add_action( 'jetpack_activate_module_pwa', 'pwa_activate' );
+add_action( 'jetpack_deactivate_module_pwa', 'pwa_deactivate' );
+
+// register WP_Query hooks for manifest and service worker
+add_filter( 'query_vars', 'pwa_register_query_vars' );
+
+// manifest
+add_action( 'wp_head', 'pwa_render_manifest_link' );
+// add_action( 'admin_head', 'pwa_render_manifest_link' ); // Don't load for wp-admin, for now
+add_action( 'amp_post_template_head', 'pwa_render_manifest_link' ); // AMP
+
+// service worker
+add_action( 'template_redirect', 'pwa_force_https', 1 );
```

Comment:
How does this work on sites that do not have an active SSL certificate?

Reply:
It doesn't work at all on those sites.